main.go

```diff
@@ -1,202 +1,272 @@
+	createClusterCmd.Flags().StringVar(&createClusterOpts.virtinkPodNetworkCIDR, "virtink-pod-network-cidr", createClusterOpts.virtinkPodNetworkCIDR, "The pod network CIDR of workload cluster")
+	createClusterCmd.Flags().StringVar(&createClusterOpts.virtinkServiceCIDR, "virtink-service-cidr", createClusterOpts.virtinkServiceCIDR, "The service CIDR of workload cluster")
+	createClusterCmd.Flags().IntVar(&createClusterOpts.virtinkMachineCPUCount, "virtink-machine-cpu-count", createClusterOpts.virtinkMachineCPUCount, "The virtink machine CPU count")
+	createClusterCmd.Flags().Var(&createClusterOpts.virtinkMachineMemorySize, "virtink-machine-memory-size", "virtink machine memory size")
+	createClusterCmd.Flags().StringVar(&createClusterOpts.virtinkMachineKernelImage, "virtink-machine-kernel-image", createClusterOpts.virtinkMachineKernelImage, "The virtink machine kernel image")
+	createClusterCmd.Flags().StringVar(&createClusterOpts.virtinkMachineRootfsImage, "virtink-machine-rootfs-image", createClusterOpts.virtinkMachineRootfsImage, "The virtink machine rootfs image")
+	createClusterCmd.Flags().Var(&createClusterOpts.virtinkMachineRootfsSize, "virtink-machine-rootfs-size", "virtink machine rootfs size")
+	rootCmd.AddCommand(createClusterCmd)
 
-			modifyKubeconfigCmd := exec.Command("kubectl", "config", "--kubeconfig", clusterKubeconfigFileName, "set-cluster", clusterName,
-				"--server", fmt.Sprintf("%s://%s:%d", u.Scheme, u.Hostname(), clusterService.Spec.Ports[0].NodePort),
-				"--tls-server-name", clusterService.Spec.ClusterIP)
-			output, err := modifyKubeconfigCmd.CombinedOutput()
-			if err != nil {
-				return fmt.Errorf("modify cluster kubeconfig: %q: %s: %s", modifyKubeconfigCmd, err, string(output))
+	rootCmd.AddCommand(&cobra.Command{
+		Use:   "delete CLUSTER",
+		Args:  cobra.ExactArgs(1),
+		Short: "Delete a workload cluster.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clusterName := args[0]
+			if err := runCommandPiped("kubectl", "delete", "clusters.cluster.x-k8s.io", clusterName, "--wait=true"); err != nil {
+				return fmt.Errorf("delete cluster template: %s", err)
 			}
 			return nil
 		},
 	})
 
 	rootCmd.AddCommand(&cobra.Command{
-		Use:  "delete CLUSTER",
-		Args: cobra.ExactArgs(1),
+		Use:   "list",
+		Short: "List workload clusters.",
 		RunE: func(cmd *cobra.Command, args []string) error {
-			clusterName := args[0]
-			if err := deleteTemplate("cluster.yaml", map[string]interface{}{"name": clusterName}); err != nil {
-				return fmt.Errorf("delete cluster template: %s", err)
+			if err := runCommandPiped("kubectl", "get", "clusters.cluster.x-k8s.io", "-o", "wide"); err != nil {
+				return fmt.Errorf("list workload clusters: %s", err)
 			}
 			return nil
 		},
 	})
-
 	if err := rootCmd.Execute(); err != nil {
 		fmt.Fprintln(os.Stderr, err)
 		os.Exit(1)
 	}
 }
 
-//go:embed manifests
-var manifests embed.FS
```

Comment:
Since the Virtink manifest is now released on GitHub, maybe we can remove it too and use the online version instead?